src/main/java/com/iggroup/universityworkshopmw/integration/controllers/OpenPositionsController.java

```diff
@@ -0,0 +1,84 @@
+package com.iggroup.universityworkshopmw.integration.controllers;
+
+import com.iggroup.universityworkshopmw.domain.exceptions.NoAvailableDataException;
+import com.iggroup.universityworkshopmw.domain.model.OpenPosition;
+import com.iggroup.universityworkshopmw.domain.services.OpenPositionsService;
+import com.iggroup.universityworkshopmw.integration.dto.OpenPositionDto;
+import com.iggroup.universityworkshopmw.integration.transformers.OpenPositionDtoTransformer;
+import com.iggroup.universityworkshopmw.integration.transformers.OpenPositionTransformer;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+import static com.iggroup.universityworkshopmw.integration.transformers.OpenPositionDtoTransformer.transform;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.http.HttpStatus.OK;
+
+@Api(value = "/openPositions", description = "Operations relating to open positions")
+@Slf4j
+@RestController
+@RequestMapping("/openPositions")
+@RequiredArgsConstructor
+public class OpenPositionsController {
+   private final OpenPositionsService openPositionsService;
+
+   @ApiOperation(value = "Get all open positions for client",
+      notes = "Returns all open positions for a give clientId",
+      response = OpenPositionDto.class,
+      responseContainer = "List")
+   @GetMapping("/{clientId}")
+   public ResponseEntity<?> getOpenPositions(@PathVariable("clientId") String clientId) {
+      try {
+         List<OpenPosition> openPositions = openPositionsService.getOpenPositionsForClient(clientId);
+         List<OpenPositionDto> responseBody = openPositions != null ? OpenPositionDtoTransformer.transform(openPositions) : null;
```

Comment:
will never be null as you handle null openPositions in getOpenPositions..() method